src/renderer/components/CollabPicker/CollabPicker.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import './CollabPicker.scss';
+import TextField from '../TextField/TextField';
+import Button from '../Button/Button';
+import Select from '@mui/material/Select';
+import { MenuItem } from '@mui/material';
+import { useState } from 'react';
+
+interface Value {
+  username: string;
+  role: number;
+}
+
+interface Options {
+  value: Value;
+  label: string;
+}
+
+interface Props {
+  value?: Value;
+  options?: Options;
+  onChange?: (...args: any[]) => void;
+}
+
+const CollabPicker: React.FC<Props> = ({ options, onChange }) => {
+  [username, onChange] = useState('');
+  const array = [{ role: value?.ro, label: string }];
```

Comment:
Based on the props a local state can be created and the `onChange` callback fired whenever the state value changes.

```tsx
const CollabPicker: React.FC<Props> = ({ value, options, onChange, onAdd }) => {
  const [currentValue, setCurrentValue] = React.useState(value);

  React.useEffect(() => {
    if (onChange === undefined) return;

    onChange(currentValue);
  }, [currentValue]);

  // ... rest of code
}
```